pywbem_mock/_mockmofwbemconnection.py

```diff
@@ -194,6 +195,34 @@ def CreateInstance(self, *args, **kwargs):
 
         return inst.path
 
+    def ModifyInstance(self, *args, **kwargs):
+        """This method is used by the MOF compiler only in the course of
+        handling CIM_ERR_ALREADY_EXISTS after trying to create an instance.
+
+        NOTE: It does NOT support the propertylist attribute that is part
+        of the CIM/XML defintion of ModifyInstance and it requires that
+        each created instance include the instance path which means that
+        the MOF must include the instance alias on each created instance.
+        """
+        mod_inst = args[0] if args else kwargs['ModifiedInstance']
+        if self.default_namespace not in self.instances:
+            raise CIMError(
+                CIM_ERR_FAILED,
+                _format('ModifyInstance failed. No instance repo exists. '
+                        'Use compiler instance alias to set path on '
+                        'instance declaration. inst: {0!A}', mod_inst))
+
+        if mod_inst.path not in self.instances[self.default_namespace]:
+            raise CIMError(
+                CIM_ERR_FAILED,
```

Comment:
Should that not result in CIM_ERR_NOT_FOUND?

Reply:
Agreed and modified and the one above it should be CIM_ERR_INVALID_NAMESPACE. DONE.